packages/vm/src/runBlock.ts

```diff
@@ -376,6 +385,42 @@ async function applyBlock(this: VM, block: Block, opts: RunBlockOpts) {
   return blockResults
 }
 
+export async function accumulateParentBlockHash(this: VM, block: Block) {
+  const historyAddress = Address.fromString(
+    bigIntToHex(this.common.param('vm', 'historyStorageAddress'))
+  )
+
+  if ((await this.stateManager.getAccount(historyAddress)) === undefined) {
+    await this.evm.journal.putAccount(historyAddress, new Account())
+  }
+
+  async function putBlockHash(this: VM, hash: Uint8Array, number: bigint) {
+    const key = setLengthLeft(bigIntToBytes(number), 32)
+    await this.stateManager.putContractStorage(historyAddress, key, hash)
+  }
+  await putBlockHash.bind(this)(block.header.parentHash, block.header.number - BIGINT_1)
+
+  // Check if we are on the fork block
+  const forkTime = this.common.eipTimestamp(2935)
+  if (forkTime === null) {
+    throw new Error('EIP 2935 should be activated by timestamp')
+  }
+  const parentBlock = await this.blockchain.getBlock(block.header.parentHash)
+
+  if (parentBlock.header.timestamp < forkTime) {
+    let ancestor = parentBlock
+    const range = this.common.param('vm', 'minHistoryServeWindow')
+    for (let i = 0; i < Number(range) - 1; i++) {
+      if (ancestor.header.number === BIGINT_0) {
+        break
+      }
+
+      ancestor = await this.blockchain.getBlock(ancestor.header.parentHash)
+      await putBlockHash.bind(this)(ancestor.hash(), ancestor.header.number)
```

Comment:
Just because I hate `.bind`, is there a reason we can't just pass the `vm` in here?  Does it make the code more clunky to do so?

Reply:
Hmm I don't know why I added this :thinking: Let me check if I can make it not use bind :)

Reply:
Ok, I think I added it because `this` as argument in your method apparently does not count as argument and therefore you need to use `bind`, I added it for reading clearlity but the method is so small, I will remove it.